src/views/Farms/components/MigrationSticky.tsx

```diff
@@ -0,0 +1,126 @@
+import React, { useState, useEffect, useRef } from 'react'
+import styled from 'styled-components'
+import { useTranslation } from 'contexts/Localization'
+import { Flex, Text, useMatchBreakpoints, WarningIcon, Button, Link } from '@pancakeswap/uikit'
+import {
+  MENU_HEIGHT,
+  MOBILE_MENU_HEIGHT,
+  TOP_BANNER_HEIGHT,
+  TOP_BANNER_HEIGHT_MOBILE,
+} from '@pancakeswap/uikit/src/widgets/Menu/config'
+
+const Container = styled.div`
+  position: sticky;
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+  width: 100%;
+  margin: auto;
+  padding: 16px;
+  z-index: 21;
+  transition: top 0.2s;
+  overflow: hidden;
+  border-bottom: 1px ${({ theme }) => theme.colors.secondary} solid;
+  border-left: 1px ${({ theme }) => theme.colors.secondary} solid;
+  border-right: 1px ${({ theme }) => theme.colors.secondary} solid;
+  border-radius: ${({ theme }) => `0 0 ${theme.radii.card} ${theme.radii.card}`};
+  background: ${({ theme }) =>
+    theme.isDark
+      ? 'linear-gradient(360deg, rgba(49, 61, 92, 0.9) 0%, rgba(61, 42, 84, 0.9) 100%)'
+      : 'linear-gradient(180deg, rgba(202, 194, 236, 0.9) 0%,  rgba(204, 220, 239, 0.9) 51.04%, rgba(206, 236, 243, 0.9) 100%)'};
+  ${({ theme }) => theme.mediaQueries.xxl} {
+    width: 1120px;
+    padding: 24px 40px;
+  }
+  ${({ theme }) => theme.mediaQueries.xl} {
+    flex-direction: row;
+  }
+`
+
+const BunnyImage = styled.img`
+  position: relative;
+  width: 210px;
+  bottom: -24px;
+  margin-left: 10px;
+
+  ${({ theme }) => theme.mediaQueries.sm} {
+    bottom: -12px;
+  }
+
+  ${({ theme }) => theme.mediaQueries.md} {
+    width: 220px;
+    right: -30px;
+    bottom: -14px;
+    margin-left: 40px;
+  }
+`
+
+const StyledLink = styled(Link)`
+  width: 100%;
+  margin-top: 8px;
+  &:hover {
+    text-decoration: initial;
+  }
+`
+
+const MigrationSticky: React.FC = () => {
+  const { t } = useTranslation()
+  const { isMobile, isXs, isSm, isMd } = useMatchBreakpoints()
+  const [stickPosition, setStickyPosition] = useState<number>(0)
+  const refPrevOffset = useRef(typeof window === 'undefined' ? 0 : window.pageYOffset)
+  const isSmallScreen = isXs || isSm || isMd
+
+  useEffect(() => {
+    const scrollEffect = () => {
+      const currentScroll = window.pageYOffset
+      if (currentScroll <= 0) {
+        setStickyPosition(0)
+        return
+      }
+      if (currentScroll >= refPrevOffset.current) {
+        setStickyPosition(0)
+      } else {
+        const warningBannerHeight = document.querySelector('.warning-banner')
+        const topBannerHeight = isMobile ? TOP_BANNER_HEIGHT_MOBILE : TOP_BANNER_HEIGHT
+        const topNavHeight = isMobile ? MENU_HEIGHT : MOBILE_MENU_HEIGHT
+        const totalTopMenuHeight = warningBannerHeight ? topNavHeight + topBannerHeight : topNavHeight
+        setStickyPosition(totalTopMenuHeight)
+      }
+      refPrevOffset.current = currentScroll
+    }
+
+    window.addEventListener('scroll', scrollEffect)
```

Comment:
We need to check window is available (i think crawlers don't have window object right @chefilip ?) before calling addEventListener

Reply:
Oh. I don't think we need to check. sometimes error throws `window.methodX` called undefined because some browsers don't support `methodX`. In this case `addEventListener` is safe because it's widely supported